Automattic/wp-calypso
# media state: introduce fetching reducer

#### Changes proposed in this Pull Request

* Introduce `media.fetching` reducer
* Remove `media.mediaRequests` reducer as its superseded by the `fetching` reducer
* Remove `requestingMedia` action because we don't need a separate action for that status

#### Testing instructions

Note: we don't currently show the `fetching` status for media items visually anywhere in calypso. To test that changes in this PR are working use the redux devtools and check state changes.

1. Open the redux tab in your browser devtools (assuming you have those installed)
2. Go to `/media/<site-id>/<media-id>`
3. Filter the action list and look out for `MEDIA_REQUEST_*` actions

Make sure your redux devtools allow to show enough actions and adjust your settings if necessary.

related to #43660
